ui/app/components/pages/create-account/import-account/json.js

```diff
@@ -82,46 +82,58 @@ class JsonImportSubview extends Component {
   }
 
   createNewKeychain () {
+    const { firstAddress, displayWarning, importNewJsonAccount, setSelectedAddress } = this.props
     const state = this.state
 
     if (!state) {
       const message = this.context.t('validFileImport')
-      return this.props.displayWarning(message)
+      return displayWarning(message)
     }
 
     const { fileContents } = state
 
     if (!fileContents) {
       const message = this.context.t('needImportFile')
-      return this.props.displayWarning(message)
+      return displayWarning(message)
     }
 
     const passwordInput = document.getElementById('json-password-box')
     const password = passwordInput.value
 
     if (!password) {
       const message = this.context.t('needImportPassword')
-      return this.props.displayWarning(message)
+      return displayWarning(message)
     }
 
-    this.props.importNewJsonAccount([ fileContents, password ])
-      // JS runtime requires caught rejections but failures are handled by Redux 
+    importNewJsonAccount([ fileContents, password ])
+      // JS runtime requires caught rejections but failures are handled by Redux
       .catch()
```

Comment:
can we do something better? not sure what. like make sure the promise is handled by something

also, wouldn't the `.catch().then()` run the `then` body after the `catch`?

Reply:
Improved in 0f3480a9